### ISSUE TITLE
Applying numpy functions on `LightCurve.time` crashes the kernel

### ISSUE THREAD

**Issue by barentsen** [KeplerGO/lightkurve#801]
Running NumPy functions on an AstroPy `Time` object accessed via `LightCurve.time` crashes my Python kernel.  For example:

```python
$ ipython
Python 3.8.2 (default, Mar 25 2020, 17:03:02) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.15.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import lightkurve as lk                               

In [2]: import astropy                                        

In [3]: import numpy as np                                    

In [4]: print(astropy.__version__)                            
4.2.dev454+gd9ad8d486

In [5]: print(np.__version__)                                 
1.18.1

In [6]: print(lk.__version__)                                 
2.0a2.dev

In [7]: lc = lk.search_lightcurve("Pi Men", sector=12).download()                                                   

In [8]: lc.time                                               
Out[8]: 
<Time object: scale='tdb' format='btjd' value=[1624.95650346 1624.95789236 1624.95928127 ... 1652.89006381 1652.8914527
 1652.89284159]>

In [9]: np.isnan(lc.time)                                     
[1]    123435 killed     ipython
```

I observe the same behavior when calling e.g. `np.mean(lc.time)`.

While I believe AstroPy `Time` objects do not currently support being passed to numpy functions, it is a major issue that this crashes the kernel rather than raising a clear error message requested the user to call e.g. `np.isnan(lc.time.value)`.

I would love for someone to try and reproduce the issue, investigate what is going on, and, if necessary, open an AstroPy issue.  I'm opening a Lightkurve issue first because it may be that our custom BKJD and BTJD time formats defined in `lightkurve/time.py` are to blame.

**Comment by dhomeier** [comment]
Essentially I can reproduce the segfault with numpy 1.18.5; with 1.19.1 it fails slightly more gracefully:
```
>>> np.isnan(lc.time)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
>>> np.mean(lc.time)
Traceback (most recent call last):
  File "/sw/lib/python3.7/site-packages/astropy/time/core.py", line 1978, in __add__
    raise OperandTypeError(self, other, '+')
astropy.time.core.OperandTypeError: Unsupported operand type(s) for +: 'Time' and 'Time'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/sw/lib/python3.7/site-packages/astropy/time/core.py", line 1967, in __add__
    if not isinstance(other, Time):
SystemError: <built-in function isinstance> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 6, in mean
  File "/sw/lib/python3.7/site-packages/numpy/core/fromnumeric.py", line 3373, in mean
    out=out, **kwargs)
  File "/sw/lib/python3.7/site-packages/numpy/core/_methods.py", line 160, in _mean
    ret = umr_sum(arr, axis, dtype, out, keepdims)
  File "/sw/lib/python3.7/site-packages/astropy/time/core.py", line 1967, in __add__
    if not isinstance(other, Time):
SystemError: <built-in function isinstance> returned a result with an error set
```
which may help a little bit further along the way. Same result with Astropys 4.0.1 and 4.1rc1; with the dev version it looks a little different – updating to the latest master already fails on `lk.search_lightcurve`:
```
>>> astropy.version.version
'4.2.dev543+geb63a7038'
>>> lc = lk.search_lightcurve("Pi Men", sector=12).download()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/derek/lib/python3.7/site-packages/lightkurve/search.py", line 581, in search_lightcurve
    campaign=campaign, sector=sector, limit=limit)
  File "/Users/derek/lib/python3.7/site-packages/lightkurve/search.py", line 727, in _search_products
    result['observation'] = None
  File "/Users/derek/lib/python3.7/site-packages/astropy/table/table.py", line 1666, in __setitem__
    self.add_column(value, name=item, copy=True)
  File "/Users/derek/lib/python3.7/site-packages/astropy/table/table.py", line 1952, in add_column
    raise ValueError('Inconsistent data column lengths')
ValueError: Inconsistent data column lengths
```
whereas running on the not-so-up-to-date version I previously had produces
```
>>> astropy.version.version
'4.2.dev407+ga049fdf50.d20200715'
>>> import lightkurve as lk
>>> lc = lk.search_lightcurve("Pi Men", sector=12).download()
>>> np.isnan(lc.time.value)
array([False, False, False, ..., False, False, False])
>>> np.isnan(lc.time)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```
so the exception looks a bit neater, but before that Python is busy for 1-2 minutes trying, and running up to ~8 GB memory usage on my system, so it's clearly not ideal. `np.mean` throws the same exception as with the older astropy version (with about the same CPU and memory usage as well).

The new `ValueError` in `Table.__setitem__` will need separate investigation...

**Comment by dhomeier** [comment]
> The new `ValueError` in `Table.__setitem__` will need separate investigation...

I could track the point of failure down to https://github.com/astropy/astropy/pull/10636 , will see when I can investigate that further to file an Astropy issue.

**Comment by barentsen** [comment]
Thanks so much @dhomeier.  Your observation of the memory leak is interesting.  I believe C extensions in Python are prone to producing a segfault if they run out of memory, so that may explain the ungraceful crash!

I played with it some more and observe the following:

The following example raises a graceful exception in a fast and memory-friendly way:
```python
t = Time(np.arange(5000), format='btjd', scale='tdb')
np.isnan(t)
```

In contrast, this example eats a ton of memory and is very slow:
```python
ts = TimeSeries(time=t)
np.isnan(ts['time'])
```

This suggests to me that the issue is specific to the use of `TimeSeries`, which is the base class upon which Lightkurve's `LightCurve` class is based.

I observed the above using the latest development branch of astropy with the following imports:
```python
from astropy.time import Time
from astropy.timeseries import TimeSeries
import numpy as np
```

I haven't been able to investigate why `np.isnan(ts['time'])` eats so much memory.  Any help in trying to diagnose this is appreciated!

**Comment by dhomeier** [comment]
Reported in https://github.com/astropy/astropy/issues/10688

**Comment by barentsen** [comment]
Thanks so much for the detailed bug report @dhomeier!  I love that you even included a plot of memory use, that's awesome!

**Comment by dhomeier** [comment]
Actually I was hoping to be able to use the debugging function, which however does not seem to respond to the memory usage – thanks for reminding me to report this!

**Comment by barentsen** [comment]
This issue has been resolved in AstroPy v4.2 via https://github.com/astropy/astropy/pull/10889.

Thanks so much for digging into this @dhomeier!
